tests/scanners/test.metadata.js

```diff
@@ -1,11 +1,24 @@
 import MetadataScanner from 'scanners/metadata';
 import * as rules from 'rules/metadata';
-import { getRuleFiles, validMetadata } from '../helpers';
+import { fakeMetadata, fakeMetadataRuleCheck, getRuleFiles,
+         validMetadata } from '../helpers';
 import { ignorePrivateFunctions } from 'utils';
 
 
 describe('MetadataScanner', () => {
 
+  it('should pass addon metadata to rules', () => {
+    var fakeRules = { fakeRule1: fakeMetadataRuleCheck };
+
+    var metadata = validMetadata();
+    var scanner = new MetadataScanner(metadata, 'fake.xpi', fakeMetadata);
+
+    return scanner.scan(fakeRules)
+      .then((validatorMessages) => {
+        assert.equal(validatorMessages.length, 0);
```

Comment:
I wonder if it might be more direct to have this be a sinon stub and check for what was passed in the assertions?